doc/lsst.pipe.base/creating-a-pipeline.rst

```diff
@@ -0,0 +1,448 @@
+the ordering of the tasks in the pipeline. The following `Pipeline` is exactly
+the same from an execution point of view.
+
+.. code-block:: yaml
+
+    description: A demo pipeline in the how-to guide
+    tasks:
+      characterizeImage: lsst.pipe.tasks.characterizeImage.CharacterizeImageTask
+      calibrate: lsst.pipe.tasks.calibrate.CalibrateTask
+      isr: lsst.ip.isr.IsrTask
+
+Tasks define their inputs and outputs which are used to construct an
+execution graph of the specified tasks. A consequence of this is if a
+pipeline does not define all the tasks required to generate all needed inputs
+and outputs it will get caught before any execution occurs.
+
+-----------------
+Configuring Tasks
+-----------------
+
+Often `PipelineTasks` (and their subtasks) contain a multitude of
+configuration options that alter the way the task executes. Because
+`Pipeline`\ s are designed to do a specific type of processing (per the
+description field) some tasks may need specific configurations set to
+enable/disable behavior in the context of a `Pipeline`.
+
+To configure a task associated with a particular label, the value associated 
+with the label must be changed from the qualified task name to a new
+sub-mapping. This new sub mapping should have two keys, ``class`` and
+``config``.
+
+The ``class`` key should point to the same qualified task name as before. The 
+value associated with the ``config`` keyword is itself a mapping where
+the configuration values are declared. The example below shows this behavior
+in action.
+
+.. code-block:: yaml
+
+  makeWarp:
+    class: lsst.pipe.tasks.makeCoaddTempExp.MakeWarpTask
+    config:
+      matchingKernelSize: 29
+      makePsfMatched: true
+      modelPsf.defaultFwhm: 7.7
+      doApplyExternalPhotoCalib: false
+      doApplyExternalSkyWcs: false
+      doApplySkyCorr: false
+      doWriteEmptyWarps: true
+
+This example shows an entry for
+`~lsst.pipe.tasks.makeCoaddTempExp.MakeWarpTask`. The label used for this task 
+is ``makeWap`` and the class location is now declared in the sub mapping
+alongside the ``class`` keyword. The ``config`` keyword is associated with 
+various `~lsst.pex.config.Field`\ s and the configuration appropriate for this 
+`Pipeline` specified as an additional yaml mapping.
+
+The complete complexity of `lsst.pex.config` can't be represented with simple
+yaml mapping syntax. To account for this, ``config`` blocks in `Pipeline`\ s
+support two special fields: ``file`` and ``python``.
+
+The ``file`` key may be associated with either a single value pointing to a
+filesystem path where a `lsst.pex.config` file can be found, or a yaml list
+of such paths. The file paths can contain environment variables that will be
+expanded prior to loading the file(s). These files will then be applied to
+the task during configuration time to override any default values.
+
+Sometimes configuration is too complex to express with yaml syntax, yet it is
```

Comment:
Yaml syntax is limited to whatever yaml supports. We have used it other places throughout the stack without linking to yaml guides so I am inclined not to. However, this may be a landing point for very new people so IDK. I am going to put it off for now, but we can always add one if we find it useful.